packages/wonder-blocks-dropdown/components/dropdown-opener.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+
+import {ClickableBehavior} from "@khanacademy/wonder-blocks-core";
+
+import type {
+    AriaProps,
+    ClickableHandlers,
+    ClickableState,
+} from "@khanacademy/wonder-blocks-core";
+
+type Props = {|
+    ...$Rest<AriaProps, {|"aria-disabled": "true" | "false" | void|}>,
+
+    /**
+     * The child function that returns the anchor the Dropdown will be activated
+     * by. This function takes `eventState`, which allows the opener element to
+     * access pointer event state.
+     */
+    children: (eventState: ClickableState) => React.Element<any>,
+
+    /**
+     * Whether the opener is disabled. If disabled, disallows interaction.
+     */
+    disabled: boolean,
+
+    /**
+     * Callback for when the opener is pressed.
+     */
+    onClick: (e: SyntheticEvent<>) => mixed,
+
+    /**
+     * Test ID used for e2e testing.
+     */
+    testId?: string,
+
+    /**
+     * Callback used to pass the childs Ref back up to the parent element
+     */
+    anchorRef: (?HTMLElement) => mixed,
+|};
+
+class DropdownOpener extends React.Component<Props> {
+    static defaultProps = {
+        disabled: false,
+    };
+
+    componentDidMount() {
+        const anchorNode = ((ReactDOM.findDOMNode(this): any): ?HTMLElement);
+        this.props.anchorRef(anchorNode);
+    }
+
+    renderAnchorChildren(
+        eventState: ClickableState,
+        handlers: ClickableHandlers,
+    ) {
+        const renderedChildren = this.props.children(eventState);
+
+        return React.cloneElement(renderedChildren, {
+            ...handlers,
+            disabled: this.props.disabled,
+            "data-test-id": this.props.testId,
```

Comment:
What happens if `children` returns an element that doesn't respond to `data-test-id`?  I feel like `ActionMenu`, `SingleSelect`, and `MultiSelect` should each pass their `testId` prop down to their own openers and we should probably remove the `testId` from `DropdownOpener`.  If people need a `testId` on their custom openers they'll need to add it themselves.

Reply:
Please correct me if I'm wrong, but I think any children should respond to `data-test-id` as we are directly setting the DOM attribute instead of a React prop. Do you have an example in mind?

Also, in that case, we could have an issue when we use the default internal opener. We wouldn't have a way to pass the `testId` to the opener DOM element.

Reply:
I don't think that we known that `renderedChildren` is a DOM node, do we?  If we use the default opener, `ActionMenu` can pass `testId` as a prop to `ActionMenuOpener` which has a `testId` prop.

Reply:
My only concern there is that people might get confused now that we have two different entrypoints for the same `testId` element (dropdown opener element). But your point is completely valid. I'm going to implement the change and we'll see how it will work when using it in different scenarios :).

Reply:
That's a valid concern.  We could require that openers have a `testId` prop.